gms/impl/src/main/java/com/linkedin/metadata/resources/entity/IngestionRunResource.java

```diff
@@ -0,0 +1,130 @@
+import com.linkedin.metadata.run.AspectRowSummary;
+import com.linkedin.metadata.run.AspectRowSummaryArray;
+import com.linkedin.metadata.run.IngestionRunSummaryArray;
+import com.linkedin.metadata.run.RollbackResponse;
+import com.linkedin.metadata.systemmetadata.SystemMetadataService;
+import com.linkedin.parseq.Task;
+import com.linkedin.restli.server.annotations.Action;
+import com.linkedin.restli.server.annotations.ActionParam;
+import com.linkedin.restli.server.annotations.Optional;
+import com.linkedin.restli.server.annotations.RestLiCollection;
+import com.linkedin.restli.server.resources.CollectionResourceTaskTemplate;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import javax.inject.Named;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * resource for showing information and rolling back runs
+ */
+@Slf4j
+@RestLiCollection(name = "runs", namespace = "com.linkedin.entity")
+public class IngestionRunResource extends CollectionResourceTaskTemplate<String, VersionedAspect> {
+
+  private static final Integer DEFAULT_OFFSET = 0;
+  private static final Integer DEFAULT_PAGE_SIZE = 100;
+  private static final Integer ELASTIC_MAX_PAGE_SIZE = 10000;
+
+  @Inject
+  @Named("systemMetadataService")
+  private SystemMetadataService _systemMetadataService;
+
+  @Inject
+  @Named("entityService")
+  private EntityService _entityService;
+
+  /**
+   * Rolls back an ingestion run
+   */
+  @Action(name = "rollback")
+  @Nonnull
+  public Task<RollbackResponse> rollback(
+      @ActionParam("runId") @Nonnull String runId,
+      @ActionParam("dryRun") @Optional @Nullable Boolean dryRun
+  ) {
+    log.info("ROLLBACK RUN runId: {} dry run: {}", runId, dryRun);
+    return RestliUtils.toTask(() -> {
+      RollbackResponse response = new RollbackResponse();
+      List<AspectRowSummary> aspectRowsToDelete;
+      aspectRowsToDelete = _systemMetadataService.findByRunId(runId);
+
+      log.info("found {} rows to delete...", stringifyRowCount(aspectRowsToDelete.size()));
+      if (dryRun) {
+        response.setAspectsAffected(aspectRowsToDelete.size());
+        response.setEntitiesAffected(aspectRowsToDelete.stream().filter(row -> row.isKeyAspect()).count());
+        response.setAspectRowSummaries(
+            new AspectRowSummaryArray(aspectRowsToDelete.subList(0, Math.min(100, aspectRowsToDelete.size())))
+        );
+        return response;
+      }
+
+      EbeanEntityService.RollbackRunResult rollbackRunResult = _entityService.rollbackRun(aspectRowsToDelete, runId);
+      List<AspectRowSummary> deletedRows = rollbackRunResult.getRowsRolledBack();
+      Integer rowsDeletedFromEntityDeletion = rollbackRunResult.getRowsDeletedFromEntityDeletion();
+
+      // since elastic limits how many rows we can access at once, we need to iteratively delete
+      while (aspectRowsToDelete.size() >= ELASTIC_MAX_PAGE_SIZE) {
+        sleep(5);
```

Comment:
Do we really need a 5 second cool off period? This seems too high. 

Also, let's make this a top-level private static variable so we can easily change it going forward

Reply:
If anything, 5 seconds is probably too conservative actually- this is waiting for MAE to handle deleting 10k elements. Will fix to make static var.

Reply:
i see.. okay